### ISSUE TITLE
height and width of videos in the video list tool

### ISSUE THREAD

**Issue by renanpwagner** [bernorieder/YouTube-Data-Tools#15]
I was wondering if you could include in the video list tool the height and width of videos so we can filter analysis for 'shorts' and 'normal' (horizontal) videos. Maybe in the generated csv file it can already classify automatically the videos in horizontal or vertical based on the width/height aspect ratio

fileDetails.videoStreams[].widthPixels | unsigned integer The encoded video content's width in pixels. You can calculate the video's encoding aspect ratio as width_pixels / height_pixels.
-- | --
fileDetails.videoStreams[].heightPixels | unsigned integer The encoded video content's height in pixels.

**Comment by bernorieder** [comment]
That's a good idea, but the fileDetails property is only available to video owners, it's not public; the only way I know of to identify shorts automatically is to probe the URL for a redirect; I will see how much load this generates and try to implement if it's not too heavy

**Comment by renanpwagner** [comment]
I assume I can probe that myself after collecting the data, right? Using the videoID info and checking if it redirects to a url with /shorts/

**Comment by bernorieder** [comment]
yes! There are different solutions discussed [here](https://stackoverflow.com/questions/71192605/how-do-i-get-youtube-shorts-from-youtube-api-data-v3), including the redirect check.

Btw, we're reading one of your papers in my YT seminar this year ,-)

**Comment by renanpwagner** [comment]
hmmm, that is very interesting. 

Didn't know you could change the channelID to get specific videos from a channel. I was trying to find a way of 'mapping' the number of shorts across time, potentially showing that more and more shorts are being produced/shared. But possibly a solution would be to get all videos, get only shorts, and then use python to match the videoIDs for shorts in the all videos data frame. 

hope they will like the paper. Which one is it?

**Comment by bernorieder** [comment]
yes, there is some cool stuff with ID tricks, you can even get lists of members-only videos. The problem with stuff like this is that it often goes away, like the type=short, mentioned in that thread, which seemed to work at one point, but does not now.

We're reading "From television to YouTube"
